internal/metrics/metrics.go

```diff
@@ -61,6 +69,20 @@ type AppProjectClientMetrics struct {
 	ProjectClientErrors prometheus.Counter
 }
 
+func NewInformerMetrics(label string) *InformerMetrics {
+	im := &InformerMetrics{
+		ResourcesListed: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "argocd_agent_informer_list_num_resources",
+			Help: "The number of resources seen by this informer",
+		}, []string{label}),
+		ListDuration: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "argocd_agent_informer_list_duration",
+			Help: "The time it took to list resources (in seconds)",
+		}, []string{label}),
+	}
+	return im
+}
```

Comment:
This slipped in, but is not used right now (except in some tests I think). It's not yet finished, either.